lib/ResolverFactory.js

```diff
@@ -10,7 +10,9 @@ const { HookMap, SyncHook, SyncWaterfallHook } = require("tapable");
 const { cachedCleverMerge } = require("./util/cleverMerge");
 
 /** @typedef {import("enhanced-resolve").Resolver} Resolver */
+/** @typedef {import("../declarations/WebpackOptions").ResolveOptions} ResolveOptions */
 
+/** @typedef {ResolveOptions & {dependencyType?: string}} EnhancedResolveOptions */
```

Comment:
The name is a bit confusing as the resolve is called `enhanced-resolve`. So `EnhancedResolveOptions` sounds like options for enhanced-resolve...